packages/web-forms/src/components/FormHeader.vue

```diff
@@ -1,17 +1,86 @@
 <script setup lang="ts">
-import Card from 'primevue/card';
-defineProps<{title: string}>();
+import { type RootNode } from '@getodk/xforms-engine';
+import PrimeButton from 'primevue/button';
+import PrimeCard from 'primevue/card';
+import PrimeMenu from 'primevue/menu';
+import { ref } from 'vue';
+import FormLanguageDialog from './FormLanguageDialog.vue';
+import FormLanguageMenu from './FormLanguageMenu.vue';
+
+const props = defineProps<{form: RootNode}>();
+const languageDialogState = ref(false);
+const menu = ref<PrimeMenu>();
+const languages = props.form.languages.filter(language => !language.isSyntheticDefault);
+
+const print = () => window.print();
+
+const items = ref([
+	{
+		label: 'Print',
+		icon: 'icon-local_printshop',
+		command: print
+	}
+]);
+
+if(languages.length > 0){
+	items.value.unshift({
+		label: 'Change language',
+		icon: 'icon-language',
+		command: () => languageDialogState.value = true
+	})
+}
 </script>
+
 <template>
-	<Card class="form-title">
+	<!-- for desktop -->
+	<div class="hidden lg:flex justify-content-end flex-wrap gap-3">
+		<PrimeButton class="print-button" severity="secondary" rounded icon="icon-local_printshop" @click="print" />
+		<FormLanguageMenu :form="form" />
+	</div>
+	<PrimeCard class="form-title hidden lg:block">
 		<template #content>
-			<h1>{{ title }}</h1>
+			<!-- TODO/q: should the title be on the definition or definition.form be accessible instead of definition.bind.form -->
+			<h1>{{ form.definition.bind.form.title }}</h1>
 			<!-- last saved timestamp -->
 		</template>
-	</Card>
+	</PrimeCard>
+
+	<!-- for mobile and tablet -->
```

Comment:
Also not something we need to act on right now (unless you want to explore it!): if we are sure we want to preserve the single-row presentation and condense horizontal space, this feels like a potentially excellent use case for [container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries). Using it well for this might have other design implications (i.e. we'd need to know how much space is needed upfront before we can query it).

The nice thing about container queries here would be the possibility of triggering space-saving as needed, which might allow more functionality to stay visible for e.g. forms with shorter titles.

Reply:
Good suggestion. For now I have run with the existing media query, I will try to use it next time we have any changes in the UI of the Form header.